erpnext/projects/doctype/task/task.py

```diff
@@ -47,9 +47,10 @@ def validate_parent_project_dates(self):
 		if not self.project or frappe.flags.in_test:
 			return
 
-		expected_end_date = getdate(frappe.db.get_value("Project", self.project, "expected_end_date"))
+		expected_end_date = frappe.db.get_value("Project", self.project, "expected_end_date")
 
 		if expected_end_date:
+			expected_end_date = getdate(expected_end_date)
 			validate_project_dates(expected_end_date, self, "exp_start_date", "exp_end_date", "Expected")
```

Comment:
Pass the argument as below
```suggestion
			validate_project_dates(getdate(expected_end_date), self, "act_start_date", "act_end_date", "Actual")
```

Reply:
there's already a validation for actual task dates below this.